engines/dfc_provider/app/services/enterprise_builder.rb

```diff
@@ -20,10 +20,20 @@ def self.enterprise(enterprise)
   end
 
   def self.enterprise_group(group)
+    members = group.enterprises.map do |member|
+      urls.enterprise_url(member.id)
+    end
+
     DataFoodConsortium::Connector::Enterprise.new(
       urls.enterprise_group_url(group.id),
       name: group.name,
       description: group.description,
-    )
+    ).tap do |enterprise|
+      # This property has been agreed by the DFC but hasn't made it's way into
+      # the Connector yet.
+      enterprise.registerSemanticProperty("dfc-b:affiliatedBy") do
```

Comment:
> Could you do this to add property not supported by the DFC onthology ?

Yes, I experimented with that already and it doesn't seem to break anything. There are some things to consider though:

* If it's not in the DFC ontology and not planned then we should probably use our own `ofn:` prefix instead of `dfc-b:`.
* The `dfc-b:` prefix is just a shortened version for a full URL. It's defined in the context. The full URL should resolve to a document which defines the ontology. We don't have our own OFN ontology and I don't know how much effort it would be to create it. So while current tools don't break, if a general purpose Semantic Web tool tried to parse it and make sense of it then it may raise errors here. So while this approach works, it's not correct until we define our own ontology with our own context.
* Using our own context may have performance impacts. Semantic Web apps cache the context and the DFC context may be preloaded while our custom context has to be defined in the document or fetched from our server.

I'm tempted to just use `ofn:customAttribute` without defining it. That's not semantic but it works when the implementing tool knows what it means.

Reply:
Nice one !
I wonder if there is a standard approach to parsing undefined context/property. If I am not mistaken browser tend to ignore tags they can't parse, so you'd think that Semantic Web tool would behave the same and just ignore unknown context/property. So I think `ofn:customAttribute` is a good place to start.

Reply:
Yes, I think that ignoring will be a very important behaviour as the standard evolves to work with new DFC versions which may not be known by the code yet. And `ofn:customAttribute` is a valid URI. It's just that the app needs to know what `ofn:` means or ignore it. :shrug: